pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -254,6 +253,16 @@ func getACRDockerEntryFromARMToken(a *acrProvider, loginServer string) (*credent
 	}, nil
 }
 
+// parseACRLoginServerFromImage takes image as parameter and returns login server of it.
+// Parameter `image` is expected in following format: foo.azurecr.io/bar/imageName:version
+// If the provided image is not an acr image, this function will return an empty string.
+func parseACRLoginServerFromImage(image string) string {
+	match := acrRE.FindAllString(image, -1)
+	if len(match) == 1 {
+		return match[0]
+	}
+	return ""
```

Comment:
Should there be a log for len(match) > 1  ?

Reply:
The image name is already logged [here](https://github.com/kubernetes/kubernetes/pull/77245/files/a2f4f51b7767612417709cdc5c174ce566970d50#diff-65860720abfa0057bc9fe725fde21bedR189). `len(match) > 1` indicates a wrong image is provided, credential provider can't handle this, in later code flow, no credential will be provided and this error will be caught outside the credential provider when trying to pull the image. Does this make sense?